### PR TITLE
DR workflow automation for CLI commands

### PR DESCRIPTION
**Pull request by olavangad-px**
**What type of PR is this?**
>integration-test

**What this PR does / why we need it**:
Adds required util functions to perform failover/failback actions and get failover/failback status.

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No

**Does this change need to be cherry-picked to a release branch?**:
Yes, 24.2.0

**Testing Details**:
Have run the tests locally.
[Jenkins job URL](https://jenkins.pwx.dev.purestorage.com/job/Users/job/omkar/job/action/2)

### CODE REVIEW

File: test/integration_test/storkctl_dr_action_test.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+func createDRAction(t *testing.T, namespace string, actionType string, migrationScheduleName string, customArgs map[string]string) (string, []string) {
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	cmdArgs := []string{"perform", actionType, "--kubeconfig", kubeConfigPath}
+	executeStorkCtlCommand(t, cmd, cmdArgs, customArgs)
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	log.InfoD("Actual output is: %s", actualOutput)
```
Comment (diptiranjanpx) [inline test/integration_test/storkctl_dr_action_test.go:185]: print command and then output of it.

File: test/integration_test/storkctl_dr_action_test.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	t.Run("createFailoverActionWithExcludeNsTest", createFailoverActionWithExcludeNsTest)
+	t.Run("createFailoverActionWithSkipSourceOperationsTest", createFailoverActionWithSkipSourceOperationsTest)
+	t.Run("createDefaultFailbackActionTest", createDefaultFailbackActionTest)
+	t.Run("createFailbackActionWithIncludeNsTest", createFailbackActionWithIncludeNsTest)
+	t.Run("createFailbackActionWithExcludeNsTest", createFailbackActionWithExcludeNsTest)
+}
+
+func createDefaultFailoverActionTest(t *testing.T) {
+	testrailId := 257169
+	actionType := "failover"
```
[the remark this answers is not in the document]
Comment (olavangad-px) [reply test/integration_test/storkctl_dr_action_test.go:50]: Will do

File: test/integration_test/storkctl_dr_action_test.go (hunk 1 above)
Comment (olavangad-px) [reply test/integration_test/storkctl_dr_action_test.go:185]: Makes sense. Will address

Comment (olavangad-px) [reply test/integration_test/storkctl_dr_action_test.go:185]: The command being executed is printed already as part of the `executeStorkCtlCommand` method

File: test/integration_test/storkctl_dr_action_test.go [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+	cmdArgs := map[string]string{
+		"migration-reference": migrationScheduleName,
+		"exclude-namespaces":  adminNs,
+		"namespace":           namespace,
+	}
+	expectedAction := generateDRActionObject(storkv1.ActionTypeFailback, migrationScheduleName, []string{defaultNs}, false)
+	createDRActionTest(t, testrailId, cmdArgs, expectedAction, migrationScheduleName, asyncDR, actionType, namespace)
+}
+
+func createDRActionTest(t *testing.T, testrailID int, args map[string]string, expectedAction *storkv1.Action, migrationScheduleName string,
```
[the remark this answers is not in the document]
Comment (olavangad-px) [reply test/integration_test/storkctl_dr_action_test.go:151]: I have created a separate createDRAction method as well.
I thought wherever else we require to only create the Action object we can use that method.
@diptiranjanpx please let me know if I did not understand your point correctly

[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/storkctl_dr_action_test.go:151]: Yes, we can use `createDRAction` for other tests.

File: test/integration_test/storkctl_dr_action_test.go [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+	t.Run("createFailoverActionWithIncludeNsTest", createFailoverActionWithIncludeNsTest)
+	t.Run("createFailoverActionWithExcludeNsTest", createFailoverActionWithExcludeNsTest)
+	t.Run("createFailoverActionWithSkipSourceOperationsTest", createFailoverActionWithSkipSourceOperationsTest)
+	t.Run("createDefaultFailbackActionTest", createDefaultFailbackActionTest)
+	t.Run("createFailbackActionWithIncludeNsTest", createFailbackActionWithIncludeNsTest)
+	t.Run("createFailbackActionWithExcludeNsTest", createFailbackActionWithExcludeNsTest)
+}
+
+func createDefaultFailoverActionTest(t *testing.T) {
+	testrailId := 257169
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/storkctl_dr_action_test.go:49]: But still there has to be these 3 tests even at end-to-end flow like
1. All Namespaces
2. A subset of namespaces.
3. A subset of namespaces + invalid namespaces.

[the remark this answers is not in the document]
Comment (olavangad-px) [reply test/integration_test/storkctl_dr_action_test.go:49]: Makes sense. Will modify / add new test cases accordingly

**Pull request merged**